### PR TITLE
[qnnpack] create a new port

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

There was no port request for this project.

This is one of the 3rd party libraries for [the PyTorch project](https://github.com/pytorch/pytorch/blob/master/.gitmodules). The PR will be used for future support of the `libtorch` port.

#### Related Works

* `nnpack` #16344: has same dependencies
* `xnnpack` #17064: has same dependencies

### Which triplets are supported/not supported? Have you updated the CI baseline?

The project's CI only considers `linux` and `osx`.

* `x64-osx`
* `x64-linux`

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@luncliff, thanks for the PR!

File: ports/qnnpack/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DQNNPACK_BUILD_TESTS=OFF
+        -DQNNPACK_BUILD_BENCHMARKS=OFF
+)
+vcpkg_install_cmake()
```
[the remark this answers is not in the document]
Comment (luncliff) [reply ports/qnnpack/portfile.cmake:19]: Thanks @PhoebeHui . Do I have to apply `vcpkg_cmake_configure` either?

[the remark this answers is not in the document]
Comment (luncliff) [reply ports/qnnpack/portfile.cmake:19]: I just tried the change and got the command is unknown. Let me check with `vcpkg-cmake`... hmm...

File: ports/qnnpack/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pytorch/QNNPACK
+    REF 7d2a4e9931a82adc3814275b6219a03e24e36b4c
+    SHA512 437a835acfedae851a9a8572fa6eea9854dcb8bcca499bc4a2582314e44f5f199778e857932da4aecf943bea7cb2eb5b1c41d4b4ca6075bddbe0f18b2c7b9127
+    HEAD_REF master
+    PATCHES
+        use-packages.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (PhoebeHui) [inline ports/qnnpack/portfile.cmake:12]:
```suggestion
vcpkg_cmake_configure(
```

File: ports/qnnpack/portfile.cmake (hunk 1 above)
Comment (PhoebeHui) [inline ports/qnnpack/portfile.cmake:19]:
```suggestion
vcpkg_cmake_install()
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@luncliff, you need add vcpkg-cmake to dependency.

You can reference similar PR https://github.com/microsoft/vcpkg/pull/17101

File: ports/qnnpack/portfile.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (luncliff) [reply ports/qnnpack/portfile.cmake]: OK. applied in 75e1f36

**Comment by luncliff** [comment]
Applied `vcpkg-cmake` in 75e1f3608181c388900420dba6187202dc64eef6. But #17063 may break this port.  
I will take care of it if this PR is merged first.

**Comment by PhoebeHui** [comment]
We should rerun this PR after PR https://github.com/microsoft/vcpkg/pull/17063 merged.

**Review by PhoebeHui — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**